ipfs-pinning-service.yaml

```diff
@@ -45,6 +45,25 @@ response. The old pin object is deleted automatically.
 
 Pin object can be removed via `DELETE /pins/{cid-of-pin-object}`
 
+# Custom metadata
+
+Pinning Services are encouraged to add support for additional features by
+leveraging optional `meta` attributes:
+
+- `PinStatus.meta[progress]` –  % progress of ongoing pinning operation
+
+- `PinStatus.meta[receivers] = ['multiaddr1','multiaddr2']` list of peers to connect to to speed up transfer of pinned data
+
+- `Pin.meta[replication]` – could define how many copies service should keep
+
+- `Pin.meta[depth]` – could define how deep the DAG should be pinned
```

Comment:
We should probably define that the default is infinite (i.e. recursive) if the server doesn't support/ignores this option. Does the server silently ignore this one too (e.g. user only wants to pin depth 1, do we pass an error along to the user "feature unsupported")?

Reply:
Can we just remove this entirely for now and go with recursive only? This could have some significant implications and ignoring this value could have some cost impact on users. I think this is outside of an MVP spec.

If the meta data can affect what's stored, we should **not** include it in this stage. I'd want to be able to 400 this request, but meta data isn't really structured right now for that.

Reply:
Fair enough. Removed `Pin.meta[depth]` from `meta` examples (d564a40eb2c0cac5326d19ddf4803a4e8c25c826).
The concept of "depth" is now gone from the spec and all pins are implicitly recursive.